connector/spanmetricsconnector/README.md

```diff
@@ -124,6 +124,8 @@ The following settings can be optionally configured:
   - `enabled`: (default: `false`): enabling will add the events metric.
   - `dimensions`: (mandatory if `enabled`) the list of the span's event attributes to add as dimensions to the events metric, which will be included _on top of_ the common and configured `dimensions` for span and resource attributes.
 - `resource_metrics_key_attributes`: Filter the resource attributes used to produce the resource metrics key map hash. Use this in case changing resource attributes (e.g. process id) are breaking counter metrics.
+- `delta_temporality`: Configuration exclusive to generating delta temporality span metrics
```

Comment:
Should I open another task that could make the configuration easier to use? The changes I have in mind are:

1. Deprecate `aggregation_temporality` and remove it from the documentation

2. Add `cumulative_temporality` to the configuration with an `enabled` property. Also move `metrics_expiration` as a property, since it's exclusively used for cumulative temporality

3. Add an `enabled` property to `delta_temporality`

4. Update `config.go` to honour both the new and old approach for picking temporality, and validate that there's no ambiguity about which temporality to use

Reply:
I would rather not introduce breaking change, like https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31106. 

we just add a new config option `metric_timestamp_cache_size` and give a proper default value

Reply:
Okay I can make that change

I was thinking it might be confusing if some configuration options are used for one temporality but not the other

Reply:
made the change